modules/consensus/difficulty.go

```diff
@@ -68,10 +68,17 @@ var (
 // however we do not use the child block deltas because that would allow the
 // child block to influence the target of the following block, which makes abuse
 // easier in selfish mining scenarios.
-func (cs *ConsensusSet) childTargetOak(parentTotalTime int64, parentTotalTarget, currentTarget types.Target, parentHeight types.BlockHeight) types.Target {
-	// Determine the detla of the current total time vs. the desired total time.
-	expectedTime := types.BlockFrequency * parentHeight
-	delta := int64(expectedTime) - parentTotalTime
+func (cs *ConsensusSet) childTargetOak(parentTotalTime int64, parentTotalTarget, currentTarget types.Target, parentHeight types.BlockHeight, parentTimestamp types.Timestamp) types.Target {
+	// Determine the delta of the current total time vs. the desired total time.
+	// The desired total time is the difference between the genesis block
+	// timestamp and the current block timestamp.
+	var expectedTime int64
+	if parentHeight < types.OakHardforkFixBlock {
+		expectedTime = int64(types.BlockFrequency * parentHeight)
+	} else {
+		expectedTime = int64(types.BlockFrequency*parentHeight) + int64(types.GenesisTimestamp)
+	}
+	delta := int64(expectedTime) - int64(parentTimestamp)
```

Comment:
This is the only change of substance in this pull request. Instead of comparing against an uncompressed timestamp, we compare against an absolute timestamp. We switch from one to the other at the block height specified by the constants.